src/dmi/icon.rs

```diff
@@ -5,7 +5,9 @@ use super::RawDmi;
 use image::imageops;
 use image::GenericImageView;
 use std::collections::HashMap;
+use std::fmt::Write as _;
```

Comment:
i don't get the need for all of this

Reply:
it's an optimization over calling format! and then calling push_str that was flagged by clippy and causing the CI to fail. Avoids an unneeded allocation.